### PR TITLE
Let dropdowns close when clicked outside of

### LINKED ISSUE #553: Make the dropdown disappear if the user clicks outside it
_From [bazsi...@gmail.com](https://code.google.com/u/118440879599163748177/) on May 03, 2013 16:25:08_

What steps will reproduce the problem? 1.Click the new campaign
2.If you click the difficulty selection list, but do not choose a difficulty and click for example on the name or on the tutorial question, the dropdown list of the difficulty remains open and does not close till you choose a difficulty 3. What is the expected output? What do you see instead? The dropdown list should automatically collapse if clicked away

What version of CorsixTH are you using (e.g. "Beta 2", "SVN r123 ")?
RC 0.21 Directx versin What operating system / compile settings are you using? Windows XP Hungarian
What level was this on (e.g. "Demo level", "Full game level 12")? Attach a saved game file (preferably an autosave from before the error) if you have one and it is relevant. Please provide any additional information below.

**Attachment:** [difficultybug.JPG](http://code.google.com/p/corsix-th/issues/detail?id=1616)

_Original issue: http://code.google.com/p/corsix-th/issues/detail?id=1616_

### PR DESCRIPTION
**Pull request by lewri**
*Fixes #553*

**Describe what the proposed change does**
- When you click outside a dropdown it will now automatically dismiss.

### CODE REVIEW

File: CorsixTH/Lua/dialogs/resizables/dropdown.lua [hunk 1]
```diff
@@ -88,3 +88,17 @@ function UIDropdown:beginDrag(x, y)
...
+--!param x (coord) x coordinate
+--!param y (coord) y coordinate
+--!return continue triggering parent function
+function UIDropdown:onMouseDown(button, x, y)
+  if not self:hitTest(x, y) then
+    self.parent_button:toggle()
+    self:close()
+    return false
+  end
+  return Window.onMouseDown(self, button, x, y)
```
Comment (lewri) [inline CorsixTH/Lua/dialogs/resizables/dropdown.lua:103]: Not sure if for consistency this should point to ``UIResizable:onMouseDown`` instead.

Comment (TheCycoONE) [reply CorsixTH/Lua/dialogs/resizables/dropdown.lua:102]: Yes it should

**Comment by TheCycoONE** [comment]
Feels natural to me, I like it :+1:

File: CorsixTH/Lua/dialogs/resizables/dropdown.lua (hunk 1 above)
Comment (lewri) [reply CorsixTH/Lua/dialogs/resizables/dropdown.lua:102]:
```suggestion
  return UIResizable.onMouseDown(self, button, x, y)
```

**Comment by lewri** [comment]
Changed, ready.

**Pull request merged**